airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -181,92 +162,27 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
 
   private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterInput connectionUpdaterInput) {
     return Workflow.newCancellationScope(() -> {
-      connectionId = connectionUpdaterInput.getConnectionId();
-
-      // workflow state is only ever set in test cases. for production cases, it will always be null.
-      if (connectionUpdaterInput.getWorkflowState() != null) {
-        workflowState = connectionUpdaterInput.getWorkflowState();
-      }
-
-      if (connectionUpdaterInput.isSkipScheduling()) {
-        workflowState.setSkipScheduling(true);
-      }
+      initInternalState(connectionUpdaterInput);
 
       // Clean the job state by failing any jobs for this connection that are currently non-terminal.
       // This catches cases where the temporal workflow was terminated and restarted while a job was
       // actively running, leaving that job in an orphaned and non-terminal state.
       ensureCleanJobState(connectionUpdaterInput);
 
-      final Duration timeToWait = getTimeToWait(connectionUpdaterInput.getConnectionId());
+      waitUntilNextRun(connectionUpdaterInput);
 
-      Workflow.await(timeToWait,
-          () -> skipScheduling() || connectionUpdaterInput.isFromFailure());
+      final boolean needToRun = needToRun(connectionUpdaterInput);
 
-      workflowState.setDoneWaiting(true);
-
-      if (workflowState.isDeleted()) {
-        log.info("Returning from workflow cancellation scope because workflow deletion was requested.");
+      if (!needToRun) {
         return;
       }
 
-      if (workflowState.isUpdated()) {
-        // Act as a return
-        prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
-      }
-
       workflowInternalState.setJobId(getOrCreateJobId(connectionUpdaterInput));
       workflowInternalState.setAttemptNumber(createAttempt(workflowInternalState.getJobId()));
 
-      final GeneratedJobInput jobInputs = getJobInput();
-
-      reportJobStarting();
-      StandardSyncOutput standardSyncOutput = null;
-
-      try {
-        final SyncCheckConnectionFailure syncCheckConnectionFailure = checkConnections(jobInputs);
-        if (syncCheckConnectionFailure.isFailed()) {
-          final StandardSyncOutput checkFailureOutput = syncCheckConnectionFailure.buildFailureOutput();
-          workflowState.setFailed(getFailStatus(checkFailureOutput));
-          reportFailure(connectionUpdaterInput, checkFailureOutput, FailureCause.CONNECTION);
-        } else {
-          standardSyncOutput = runChildWorkflow(jobInputs);
-          workflowState.setFailed(getFailStatus(standardSyncOutput));
-
-          if (workflowState.isFailed()) {
-            reportFailure(connectionUpdaterInput, standardSyncOutput, FailureCause.UNKNOWN);
-          } else {
-            reportSuccess(connectionUpdaterInput, standardSyncOutput);
-          }
-        }
+      final GeneratedJobInput jobInputs = generateJobInput(connectionUpdaterInput);
```

Comment:
I would remove `generateJobInput` and directly call `getJobInput` or remove line 180:181

Reply:
same thought. Might be missing something but doesn't seem useful to wrap this up into a separate function.

Reply:
Done, I also added a method to create the Job and The attempt